GetBPDTransactions/handler.ts

```diff
@@ -104,21 +108,28 @@ export function GetBPDTransactionsHandler(
 export function GetBPDTransactions(
   citizenRepository: TaskEither<Error, Repository<Transaction>>,
   insertOrReplaceEntity: InsertOrReplaceEntity,
-  publicRsaCertificate: NonEmptyString
+  publicRsaCertificate: NonEmptyString,
+  adb2cCreds: IServicePrincipalCreds,
+  adb2cAdminGroup: NonEmptyString,
+  cacheTtl: NumberFromString
 ): express.RequestHandler {
   const middlewaresWrap = withRequestMiddlewares(
     ContextMiddleware(),
-    RequiredExpressUserMiddleware(AdUser),
-    RequestCitizenToFiscalCode(publicRsaCertificate)
+    RequestCitizenToFiscalCode(
+      publicRsaCertificate,
+      adb2cCreds,
+      adb2cAdminGroup,
+      cacheTtl
+    )
   );
 
   const handler = withAudit(insertOrReplaceEntity)(
     GetBPDTransactionsHandler(citizenRepository),
-    (context, { oid }, fiscalCode) => ({
-      AuthLevel: "Admin",
+    (context, { user, fiscalCode }) => ({
+      AuthLevel: isAdminAuthLevel(user, adb2cAdminGroup) ? "Admin" : "Support",
       Citizen: fiscalCode,
       OperationName: "GetBPDCitizen",
-      PartitionKey: oid, // Can we use email?
+      PartitionKey: user.oid, // Can we use email?
       RowKey: context.executionContext.invocationId as string & NonEmptyString
```

Comment:
My only concern about it is that can't be traced when a user uses his privileges to query by `fiscalCode` instead `supportToken`. This information into the audit could monitor the abusing of Admin privileges.

Reply:
I think that if a user has admin privileges maybe he always will prefer to use fiscal code instead of support token, while querying citizen data, so why talking about abusing?

Reply:
He shouldn't. fiscalCode should be used only in well defined contexts and justified. Say when there isn't a support token, it isn't working or other use cases that weren't capture in the functional analysis.

Reply:
In this case we have to modify our middleware properly to reach this info. Can we move this into another PR? @BurnedMarshal @pp-ps

Reply:
Nevermind, inserted in [49dcc41](https://github.com/pagopa/io-functions-backoffice/pull/17/commits/49dcc41c84bdaf65068f30bdf438d7674f975d80)